discovery_engine_core/core/src/engine.rs

```diff
@@ -581,6 +572,55 @@ where
         }
     }
 
+    /// Performs a deep search regarding a document.
+    pub async fn deep_search(
+        &self,
+        snippet: &str,
+        market: &Market,
```

Comment:
I think we can because even if a provider does not return we know from which market it came from and we can set it before returning the articles.
I think this is form is to be preferred because it requires simple data to work and not a full `Document`.